cf-agent/verify_files_utils.c

```diff
@@ -3243,14 +3244,22 @@ static void SaveSetxid(bool modified)
     {
         char filename[CF_BUFSIZE];
         snprintf(filename, CF_BUFSIZE, "%s/cfagent.%s.log", GetLogDir(), VSYSNAME.nodename);
+        ToLowerStrInplace(filename);
         MapName(filename);
 
         PurgeItemList(&VSETXIDLIST, "SETUID/SETGID");
 
         Item *current = RawLoadItemList(filename);
         if (!ListsCompare(VSETXIDLIST, current))
         {
+            mode_t oldmode = umask(077); // This setxidlist file must only be accesible by root
+            Log(LOG_LEVEL_DEBUG, "Updating setxidlist at '%s', current umask is %o, will create/update setxidlist using 0600.", filename, oldmode);
+            if (safe_chmod(filename, 0600)) // need to chmod in case the file already exists
+            {
```

Comment:
@olehermanse I am now wondering why I bother with safe_chmod() since the RawSaveItemList() call will always write a new file with the current umask. I tested this locally and things work well without safe_chmod().